File: frontend/models/Places.ts
import type { ServiceWithTimetable } from "./ServiceInfo";
import type { Stop } from "./Stop";

export interface Locality {
    id: string;
    name: string;
    qualifier?: string;
    full_name?: string;
}

export interface LocalityDetails extends Locality {
    services: ServiceWithTimetable[];
    stops: Stop[];
}

export interface District {
    id: string;
    name: string;
}

export interface DistrictDetails extends District {
    localities: Locality[];
}

export interface AdminArea {
    id: string;
    name: string;
}

export interface AdminAreaDetails extends AdminArea {
    districts: District[];
}

export interface Region {
    id: string;
    name: string;
}

export interface RegionDetails extends Region {
    admin_areas: AdminAreaDetails[];
}

File: backend/schemas/places.py
from pydantic import BaseModel
from typing import Optional
from backend.schemas.service import Service
from backend.schemas.stop import Stop


class Locality(BaseModel):
    id: str
    name: str
    qualifier: Optional[str] = None
    full_name: Optional[str] = None


class LocalityDetails(Locality):
    services: list[Service]
    stops: list[Stop]


class District(BaseModel):
    id: str
    name: str


class DistrictDetails(District):
    localities: list[Locality]


class AdminArea(BaseModel):
    id: str
    name: str


class AdminAreaDetails(AdminArea):
    districts: list[District]


class Region(BaseModel):
    id: str
    name: str


class RegionDetails(Region):
    admin_areas: list[AdminArea]
